integration/conftest.py

```diff
@@ -75,7 +75,10 @@ def _delete_unused_network_interface_by_subnet(ec2_client, subnet_id):
         network_interface_ids += [ni["NetworkInterfaceId"] for ni in page["NetworkInterfaces"]]
 
     for ni_id in network_interface_ids:
-        ec2_client.delete_network_interface(NetworkInterfaceId=ni_id)
+        try:
+            ec2_client.delete_network_interface(NetworkInterfaceId=ni_id)
```

Comment:
Shall we do the detach first before deleting? `You must detach the network interface before you can delete it.`

Reply:
There is a check above that checks whether the status is `available`. And `if the network interface is not attached to an instance, the status is available`